### PR TITLE
Fix Issue 22782 - address of ref can be assigned to non-scope parameter

### PR DESCRIPTION
**Pull request by dkorpel**

### CODE REVIEW

**Comment by dukc** [comment]
Does this by any chance also fix https://issues.dlang.org/show_bug.cgi?id=22783? It might have the same underlying cause.

**Comment by dukc** [comment]
1 hour and 12 minutes from issue report to fix PR. Very impressive!

**Comment by dkorpel** [comment]
> Does this by any chance also fix https://issues.dlang.org/show_bug.cgi?id=22783? It might have the same underlying cause.

Yes, I added the test case and closed it as duplicate.

**Comment by dkorpel** [comment]
> 1 hour and 12 minutes from issue report to fix PR. Very impressive!

Thanks! But it's still blocked as Phobos needs to be updated, looks like std.bigint and std.file on Windows relied on this bug.

File: test/fail_compilation/test20245.d [hunk 1]
```diff
@@ -41,3 +53,35 @@ void main() @safe
...
+
+// argument is not, or should not be scope
+auto listUp(const(char)* content) {charPtr = content;}
+
+struct DontDoThis2
+{
+    char content;
+    @safe escape()
+    {
+        // should not compile, but does.
```
Comment (dukc) [inline test/fail_compilation/test20245.d:84]: Leave this comment off. Or at least that "but does" - part. It's out of date.

**Comment by thewilsonator** [comment]
Dependant PR merged.

**Comment by dkorpel** [comment]
It's green now

File: smallbuffer.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+dmd/common/string SmallBuffer
```
Comment (MoonlightSentinel) [inline smallbuffer.txt:1]: Why was this file created? It's completely unrelated to the PR

Comment (dkorpel) [reply smallbuffer.txt:1]: Wait, it's still here? This should be deleted 😅

Comment (MoonlightSentinel) [reply smallbuffer.txt:1]: #13698

**Pull request merged**